### ISSUE TITLE
.ps1 shims breaking applications that call program via start-process

### ISSUE THREAD

**Issue by JustinGrote** [ScoopInstaller/Scoop#3749]
Simple one:
scoop install git
Invoke-Git

`Exception calling "Start" with "0" argument(s): "The specified executable is not a valid application for this OS platform."`

This happens because the program looks for "git" and "git.ps1" is returned as the first result, which is not an exe.

Is there a way to make git.exe the first in the path search? maybe two paths for shims with the .ps1 in a lower order?

Is there a setting to disable .ps1 shim creation? Why do they exist anyways if the .exe shims are created?

**Comment by mihai-unity** [comment]
Yeah, and:

```ps1
PS C:\> (Get-Command "python").Path
C:\Users\mihai\scoop\shims\python.ps1
``` 

will break any subsequent `Start-Process` relaying on that information:
```
Start-Process : This command cannot be run due to the error: %1 is not a valid Win32 application.
```

**Comment by mihai-unity** [comment]
FWIW, I'm going to do this:

```ps1
$pythonPath = (Get-Command "python").Path
if ((Get-Item $pythonPath).Extension -eq ".ps1") {
    $pythonPath = iex "$pythonPath -c `"import sys; print(sys.executable)`""
}
```

**Comment by JustinGrote** [comment]
@r15ch13 would you accept a pull request to allow a manifest option to suppress the .ps1 generation for apps where this is a continual problem?

**Comment by imgx64** [comment]
Another issue caused by .ps1 shims: mklement0/Native#3

**Comment by rashil2000** [comment]
Merged in develop branch - #4543
